install_with_binary.sh

```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+
+# Function to display step messages
+display_step() {
+    echo "Step $1: $2"
+}
+
+# Step 1: Detect the current OS and Architecture type
+display_step 1 "Detecting current OS and Arch "
+os=""
+case "$(uname -sr)" in
+   Darwin*)
+     os="darwin" ;;
+   Ubuntu*|Pop!_OS)
+     os="linux" ;;
+   Linux*amzn2*)
+     os="amazon linux" ;;
+   Linux*)
+     os="linux" ;;
+   Debian*)
+     os="linux" ;;
+   Linux\ Mint*)
+     os="linux" ;;
+   Red\ Hat*)
+     os="linux" ;;
+   CentOS*)
+     os="linux" ;;
+   *)
+     os="Not Found: $os_name"
+     echo 'Not Supported OS'
+    exit 1
+     ;;
+esac
+
+arch=$(uname -m)
+if [[ $arch == x86_64* ]]; then
+    arch="amd64"
+elif  [[ $arch == arm* || $arch == "aarch64" ]]; then
+    arch="arm64"
+else
+    echo 'Not Supported Architecture'
+fi
+
+# Step 2: Detect the latest version from siglens release
+display_step 2 "Getting lastest version of siglens release"
+
+latest_version=`\
+    curl  --silent "https://api.github.com/repos/siglens/siglens/releases/latest" |
+    grep '"tag_name":' |
+    sed -E 's/.*"([^"]+)".*/\1/'`
+
+# Step 3: Fetching latest binary based on OS and Arch
+display_step 3 "fetching latest binary for $os-$arch and version $latest_version"
```

Comment:
nit: `fetching` is not capitalized, but the first word of your other Step messages are capitalized